### PR TITLE
Atoms rebased unsquashed

### PR DESCRIPTION
**Pull request by bantic**
rebased/squashed/unsquashed version of #226, with tests for the cases mentioned in https://github.com/bustlelabs/mobiledoc-kit/pull/226#issuecomment-173630690

**Ready for review**

cc @mixonic @rlivsey

### CODE REVIEW

**Comment by rlivsey** [comment]
:+1: awesome stuff

File: src/js/editor/key-commands.js [hunk 1]
```diff
@@ -53,6 +53,26 @@ export const DEFAULT_KEY_COMMANDS = [{
...
+  run(editor) {
+    let range = editor.cursor.offsets;
+    let {head: {section}} = range;
+    editor.run(postEditor => {
+      postEditor.setRange(new Range(section.headPosition()));
+    });
+  }
+}, {
+  // FIXME restrict to OS X only?
+  str: 'CTRL+E',
```
Comment (mixonic) [inline src/js/editor/key-commands.js]: This is a new feature? Or something that broke?

File: src/js/models/image.js [hunk 2]
```diff
@@ -12,11 +11,7 @@ export default class Image extends Section {
     return false;
   }
 
-  headPosition() {
-    return new Position(this, 0);
-  }
-
-  tailPosition() {
-    return new Position(this, 1);
+  get length() {
```
Comment (mixonic) [inline src/js/models/image.js]: IMO this whole model should go the way of the Dodo before 1.0. images are just cards.

File: src/js/editor/key-commands.js (hunk 1 above)
Comment (bantic) [inline src/js/editor/key-commands.js]: The ctr-A and ctr-E OS X native movement commands don't work when there are atoms in the text (the cursor focus gets moved to the inside of the atom). I have a branch will PR after this that addresses these FIXMEs and scopes to Mac only.

File: src/js/models/image.js (hunk 2 above)
Comment (bantic) [inline src/js/models/image.js]: 100% agree

File: ATOMS.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+The return value of the `render` function will be inserted by the renderer into the rendered mobiledoc.
+The return value can be null if an atom does not have any output. If there is a return value it
+must be of the correct type (a DOM Node for the dom renderer, a string of html or text for the html or text renderers, respectively).
+
+#### `env`
+
+`env` always has the following properties:
+
+  * `name` [string] - the name of the card
```
Comment (praxxis) [inline ATOMS.md]: name of the card or atom? Atoms are said to be effectively cards, but I don't read that as "atom and card are interchangeable". We also use `card` as the var name in the example on line 49

Comment (bantic) [inline ATOMS.md]: Thanks @praxxis, you're right. Will fix this.

**Pull request merged**